MedvedevKS/lab_5/lab_5.cpp

```diff
@@ -0,0 +1,219 @@
+#include "stdafx.h"
+#include <conio.h>
+#include <iostream>
+using namespace std;
+
+const int M = 49, N = 50;
+
+void MatrixSum(int* M1, int* M2, int* M3, int m, int n)
+{
+	for (int i = 0; i < m; i++)
+	{
+		for (int j = 0; j < n; j++)
+		{
+			*(M3 + i*N + j) = *(M1 + i*N + j) + *(M2 + i*N + j);
+		}
+	}
+}
+
+void MatrixSubstraction(int* M1, int* M2, int* M3, int m, int n)
+{
+	for (int i = 0; i < m; i++)
+	{
+		for (int j = 0; j < n; j++)
+		{
+			*(M3 + i*N + j) = *(M1 + i*N + j) - *(M2 + i*N + j);
+		}
+	}
+}
+
+void MatrixProduct(int* M1, int* M2, int* M3, int m1, int common, int n2)
+{
+	for (int i = 0; i < m1; i++)
+	{
+		for (int j = 0; j < n2; j++)
+		{
+			int k = 0;
+			for (int q = 0; q < common; q++)
+			{
+				*(M3 + i*N + j) += *(M1 + i*N + k) * *(M2 + k*N + j);
+				k++;
+			}
+		}
+	}
+}
+
+int main()
+{
+	char end;
+	cout << "MaTrIxCaLc\n\n";
+	do
+	{
+		int m1, n1, m2, n2, m3, n3;
+		int MatrixA[M][N], MatrixB[M][N];
+		int* p1 = &MatrixA[0][0];
+		int* p2 = &MatrixB[0][0];
+		cout << "Enter matrix A size (m1 n1)\n";
+		cin >> m1 >> n1;
+		cout << "Define matrix A\n\n";
+		for (int i = 0; i < m1; i++)
+		{
+			for (int j = 0; j < n1; j++)
+			{
+				cin >> *(p1 + i*N + j);
+			}
+		}
+		cout << "\n\nEnter matrix B size (m2 n2)\n";
+		cin >> m2 >> n2;
+		cout << "Define matrix B\n\n";
+		for (int i = 0; i < m2; i++)
+		{
+			for (int j = 0; j < n2; j++)
+			{
+				cin >> *(p2 + i*N + j);
+			}
+		}
+		do
+		{
+			int MatrixRes[M][N] = { 0 };
+			int* p3 = &MatrixRes[0][0];
+			cout << "\nWhat do u want to do with matrices? Possible operations:\n";
+			bool result;
+			if (m1 == m2 && n1 == n2 && n1 == m2)
```

Comment:
Extract each situation (m1 == m2 && n1 == n2 && n1 == m2, n1 == m2 && m1 == n2 and so on) to separate function.